src/communication/communication_channel.cpp

```diff
@@ -0,0 +1,297 @@
+{
+  this->~communication_channel ();
+
+  new (this) communication_channel (std::forward <communication_channel> (comm));
+  return *this;
+}
+
+communication_channel::~communication_channel ()
+{
+  close_connection ();
+  free (iov_ptr);
+}
+
+int communication_channel::send ()
+{
+  int total_len = 0;
+  int rc = 0, vector_length = 0;
+
+  for (unsigned int i = 0; i < current_iov_pointer; i++)
+    {
+      total_len += iov_ptr[i].iov_len;
+    }
+
+  vector_length = current_iov_pointer;
+  current_iov_pointer = 0;
+
+  while (total_len > 0)
+    {
+      rc = css_vector_send (m_socket, &iov_ptr, &vector_length, rc, m_max_timeout_in_ms);
+      if (rc < 0)
+	{
+	  close_connection ();
+	  return ERROR_ON_WRITE;
+	}
+      total_len -= rc;
+    }
+
+  return NO_ERRORS;
+}
+
+int communication_channel::send (const std::string &message)
+{
+  return communication_channel::send (message.c_str (), message.length ());
+}
+
+int communication_channel::recv (char *buffer, int &received_length)
+{
+  return css_net_recv (m_socket, buffer, &received_length, m_max_timeout_in_ms);
+}
+
+int communication_channel::connect ()
+{
+  int length = 0;
+  int rc = NO_ERRORS;
+
+  assert (m_type == CHANNEL_TYPE::INITIATOR);
+  assert (m_command_type > NULL_REQUEST && m_command_type < MAX_REQUEST);
+
+  if (m_type != CHANNEL_TYPE::INITIATOR || is_connection_alive ())
+    {
+      return ER_FAILED;
+    }
+
+  if (m_server_name == nullptr)
+    {
+      length = 0;
+    }
+  else
+    {
+      length = strlen (m_server_name.get ());
+    }
+
+  m_socket = css_tcp_client_open (m_hostname.get (), m_port);
+  if (!IS_INVALID_SOCKET (m_socket))
+    {
+      /* send magic */
+      NET_HEADER magic_header = DEFAULT_HEADER_DATA, command_header = DEFAULT_HEADER_DATA, data_header = DEFAULT_HEADER_DATA;
+      memset ((char *) &magic_header, 0, sizeof (NET_HEADER));
+      memcpy ((char *) &magic_header, css_Net_magic, sizeof (css_Net_magic));
+
+      rc = communication_channel::send ((const char *) &magic_header, sizeof (NET_HEADER));
+      if (rc != NO_ERRORS)
+	{
+	  close_connection ();
+	  return REQUEST_REFUSED;
+	}
+
+      /* send request */
+      css_set_net_header (&command_header, COMMAND_TYPE, m_command_type, m_request_id, length, 0, 0, 0);
+      if (length > 0)
+	{
+	  css_set_net_header (&data_header, DATA_TYPE, 0, m_request_id, length, 0, 0, 0);
+	  rc = communication_channel::send ((char *) &command_header, sizeof (NET_HEADER), (char *) &data_header,
+					    sizeof (NET_HEADER), m_server_name.get (), length);
+	  if (rc != NO_ERRORS)
+	    {
+	      close_connection ();
+	      return REQUEST_REFUSED;
+	    }
+	}
+      else
+	{
+	  rc = communication_channel::send ((char *) &command_header, sizeof (NET_HEADER));
+	  if (rc != NO_ERRORS)
+	    {
+	      close_connection ();
+	      return REQUEST_REFUSED;
+	    }
+	}
+
+      _er_log_debug (ARG_FILE_LINE, "connect_to_master:" "connected to master_hostname:%s\n", m_hostname.get ());
```

Comment:
Remove forced logging by either:

  1. use er_log_debug (is activate only when debug logging is set).
  2. enable logging by system parameter.